torch_xla/distributed/xla_dist.py

```diff
@@ -86,14 +102,39 @@ def __init__(self,
                ' will interfere with the values set for distributed'
                ' training'.format(dist_var)))
 
+  def _check_client_mesh_health(self, health_timeout=120):
+    max_delay = 0
```

Comment:
This should be different.
```Python
now = time.time()
lag = 0.0
count = None
for h in hbeats:
  lag = max(lag, h.last_time)
  if count is None:
    count = h.count
  elif count >= 0 and count != h.count:
    cont = -1
if count < 0 and lag > MAX_LAG_UNEVEN:
  raise ...
if count > 0 and lag > MAX_LAG_EVEN:
  raise ... 
```

Then I'd use MAX_LAG_EVEN=15..30m and MAX_LAG_UNEVEN=5m.

Reply:
Let's have a chat ... :smile:

Reply:
Updated the health check logic to toggle between even and uneven cases following offline discussion.